designer/server/src/config.ts

```diff
@@ -1,130 +1,167 @@
-    .default('debug'),
-  phase: joi.string().valid('alpha', 'beta').optional(),
+  logLevel: joi.string().valid('trace', 'info', 'debug', 'error'),
+  phase: joi.string().valid('alpha', 'beta', 'live'),
   footerText: joi.string().optional(),
-  lastCommit: joi.string().default('undefined'),
-  lastTag: joi.string().default('undefined'),
-  sessionTimeout: joi.number().default(sessionSTimeoutInMilliseconds),
-  sessionCookieTtl: joi.string().optional(),
-  sessionCookiePassword: joi.string().optional(),
+  isProduction: joi.boolean().default(false),
+  isDevelopment: joi.boolean().default(true),
+  isTest: joi.boolean().default(false),
+  lastCommit: joi.string().optional(),
+  lastTag: joi.string().optional(),
+  sessionTtl: joi.number(),
+  sessionCookieTtl: joi.number(),
+  sessionCookiePassword: joi.string(),
+  awsCredentials: joi
+    .object()
+    .keys({
+      accessKeyId: joi.string(),
+      secretAccessKey: joi.string(),
+      sessionToken: joi.string().optional()
+    })
+    .optional(),
   azureClientId: joi.string().optional(),
   azureClientSecret: joi.string().optional(),
   oidcWellKnownConfigurationUrl: joi.string().optional(),
-  appBaseUrl: joi.string().optional().default('http://localhost:3000'),
-  useSingleInstanceCache: joi.string().optional().default(false),
+  appBaseUrl: joi.string(),
   redisHost: joi.string(),
   redisUsername: joi.string(),
   redisPassword: joi.string(),
-  redisKeyPrefix: joi.string().optional().default('forms-designer'),
-  redisTtl: joi.number().optional().default(2419200000), // one day
-  serviceName: joi.string()
+  redisKeyPrefix: joi.string()
 })
 
+const {
+  APP_BASE_URL = 'http://localhost:3000',
+  APP_PATH_PREFIX = '/forms-designer',
+  AZURE_CLIENT_ID,
+  AZURE_CLIENT_SECRET,
+  FOOTER_TEXT,
+  LAST_COMMIT,
+  LAST_COMMIT_GH,
+  LAST_TAG,
+  LAST_TAG_GH,
+  LOG_LEVEL = 'info',
+  NODE_ENV = 'development',
+  OIDC_WELL_KNOWN_CONFIGURATION_URL,
+  PERSISTENT_BACKEND = 'preview',
+  PHASE = 'beta',
+  PORT = '3000',
+  PREVIEW_URL = 'http://dev.cdp-int.defra.cloud/forms-runner/',
+  PUBLISH_URL = 'http://dev.cdp-int.defra.cloud/forms-runner/',
+  REDIS_HOST = 'localhost',
+  REDIS_KEY_PREFIX = 'forms-designer',
+  REDIS_PASSWORD = 'my-password',
+  REDIS_USERNAME = 'default',
+  SESSION_COOKIE_PASSWORD = '',
+  SESSION_COOKIE_TTL,
+  SESSION_TTL,
+  S3_BUCKET
+} = process.env
+
 // Build config
-const config = {
-  port: process.env.PORT,
-  env: process.env.NODE_ENV,
-  appPathPrefix: process.env.APP_PATH_PREFIX || '/forms-designer',
-  previewUrl:
-    process.env.PREVIEW_URL || 'http://dev.cdp-int.defra.cloud/forms-runner/', // TODO set this to localhost and pull env vars from CDP
-  publishUrl:
-    process.env.PUBLISH_URL || 'http://dev.cdp-int.defra.cloud/forms-runner/', // TODO set this to localhost and pull env vars from CDP
-  persistentBackend: process.env.PERSISTENT_BACKEND || 'preview',
+const config: Config = {
+  port: parseInt(PORT),
```

Comment:
Better to let `joi` do the type coercion from ENVAR strings (happens during `schema.validate`).
It might have to stay for while though to keep TS happy?